server/executor/poller_executor.go

```diff
@@ -113,7 +121,14 @@ func (pe DefaultPollerExecutor) donePollingTraces(job *PollingRequest, trace tra
 		return false
 	}
 
-	if len(trace.Flat) > 0 && len(trace.Flat) == len(job.run.Trace.Flat) {
+	// We always expect to get the tracetest trigger span, so it has to have more than 1 span
+	if len(trace.Flat) > 1 && len(trace.Flat) == len(job.run.Trace.Flat) {
+		pe.repeatedTraceSizeCache[trace.ID.String()] = pe.repeatedTraceSizeCache[trace.ID.String()] + 1
+	} else {
+		pe.repeatedTraceSizeCache[trace.ID.String()] = 0
+	}
+
+	if pe.repeatedTraceSizeCache[trace.ID.String()] >= 3 {
```

Comment:
This way, we only return true if the number of spans doesn't change after 3 pollings

Reply:
Decided to remove it from this PR. This solution is too raw to be merged, I'll work on a better approach later.